dist/zone.js

```diff
@@ -1570,6 +1570,9 @@ function patchXHR(window) {
         return abortNative.apply(data.target, data.args);
     }
     var openNative = patchMethod(window.XMLHttpRequest.prototype, 'open', function () { return function (self, args) {
+        //to handle async if it is being used with any other library or windows.XMLHttpRequest.prototype with other library's
+        if(args[2]===undefined)
```

Comment:
could you provide a sample code or repo to describe the error scenario ? 
If the args[2] is undefined, the original code 

```javascript
         self[XHR_SYNC] = args[2] == false;
```

will return false, that means this request will be an async call, I think the original code is correct behavior, your code is not correct to me. because open(url, options, async[options]), by default the async will be true.

about the "More task than scheduled " error, maybe you can check my PR #529 to check whether it fix your issue or not.

And BTW you can't update dist/zone.js directly, you should update lib/browser/browser.ts in  this case.